libraries/ms-common/src/x86/gpio.c

```diff
@@ -29,10 +29,12 @@ static uint32_t prv_get_index(const GPIOAddress *address) {
 }
 
 StatusCode gpio_init(void) {
-  GPIOSettings default_settings = { .direction = GPIO_DIR_IN,
-                                    .state = GPIO_STATE_LOW,
-                                    .resistor = GPIO_RES_NONE,
-                                    .alt_function = GPIO_ALTFN_NONE };
+  GPIOSettings default_settings = {
+    .direction = GPIO_DIR_IN,
```

Comment:
There are too many arguments to put this on one line so it did one per line as per the rule. Basically, if it can pack the initializer arguments into a single line it will. If it can't one per line. 

There is a fairly thorough discussion of the  problem and its solutions [here](https://stackoverflow.com/questions/33656800/clang-format-line-breaks).

TL;DR: Solution are
* Use `\\ clang-format [off | on]` comments.
* Add `\\` after each parameter to force a line break.
* Set `ColumnLimit: 0` which disables wrapping but won't mess with initializers (although you must manually format them the first time.

I'm not a fan of any of those solutions and would rather go with its present behavior. IMO it is consistent once you understand the too and while the one line is ever so slightly less readable I don't see significant value in not using the tool for this single problem.